### DISCUSSION TITLE
TLS Handshake failed with error -7780

### DISCUSSION THREAD

**Discussion by vipindas1984** [zephyrproject-rtos/zephyr#55052] [Q&A]
I am trying to setup a thread network(nRF5340 DK & RPi)  to send  RPC message to a remote http server:https://eth-mainnet.g.alchemy.com/v2/B8W2IZrDkCkkjKxQOl70XNIy4x4PT20S). I took http_client sample(sdk-zephyr) as a reference. I was successful in sending RPC message to the server via TCP socket connection but when  I tried to communicate over TLS socket, I am getting TLS Handshake error(-7780).  

> [00:00:43.134,155] <err> mbedtls: ssl_msg.c:4494: is a fatal alert message (msg 40)
[00:00:43.134,277] <err> mbedtls: ssl_msg.c:3604: mbedtls_ssl_handle_message_type() returned -30592 (-0x7780)
[00:00:43.134,368] <err> mbedtls: ssl_cli.c:1924: mbedtls_ssl_read_record() returned -30592 (-0x7780)
[00:00:43.134,399] <wrn> mbedtls: ssl_tls.c:5323: <= handshake
[00:00:43.134,429] <err> net_sock_tls: TLS handshake error: -7780
[00:00:43.138,397] <err> http_client: Cannot connect to IPv6 remote (-113)
[00:00:43.138,427] <err> http_client: Establishing socket connection failed
[00:00:43.138,427] <err> http_client: Setup socket connection failed

Can someone tell me what could be the possible reason for this error. I had taken a look at different discussions on similar issues and also asked my questions on Devzone etc. But didn't get much helpful answer. 

> Zephyr-sdk: 0.15.1
Zephyr Version: 3.2.99

Regards

Vipin Das

**Comment by GeorgePriestner-Eaton** [comment]
@vipindas1984 Did you find any resolution to this issue?

**Reply by vipindas1984** [reply]
Yes, most of my issues got resolved. The issue got resolved after I updated the nrf connect sdk version to v2.2.99-dev3. 

You can try the same and let me know if you still have issues.

**Comment by vipindas1984** [comment]
Hi George,

Yes, most of my issues got resolved. The issue got resolved after I updated
the nrf connect sdk version to v2.2.99-dev3.

You can try the same and let me know if you still have issues.

Thanks

Vipin Das

On Fri, May 26, 2023 at 5:57 PM GeorgePriestner-Eaton <
***@***.***> wrote:

> @vipindas1984 <https://github.com/vipindas1984> Did you find any
> resolution to this issue?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zephyrproject-rtos/zephyr/discussions/55052#discussioncomment-6013464>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AH34VCMSFFK7N4VLA6PJ3ULXIDHFHANCNFSM6AAAAAAVDSIBIE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
Vipin Das B
